craft_platforms/_architectures.py

```diff
@@ -98,4 +100,7 @@ def parse_base_and_architecture(
         base = None
         arch_str = arch
 
-    return base, DebianArchitecture(arch_str) if arch_str != "all" else "all"
+    try:
+        return base, DebianArchitecture(arch_str) if arch_str != "all" else "all"
+    except ValueError as err:
+        raise ValueError(f"{arch_str!r} is not a valid architecture.") from err
```

Comment:
```suggestion
        raise ValueError(f"{arch_str!r} is not a valid Debian architecture.") from err
```

Also maybe this should be raise `from None`

Reply:
> Also maybe this should be raise from None

Because it doesn't add any useful information to the traceback?

Reply:
the two almost-identical ValueErrors seem redundant on the traceback, yea